### PR TITLE
Merging initial commits to master

### PR DESCRIPTION
**Pull request by adrianodennanni**
This PR has all functionalities to solve image captchas, and documentation is included.

### CODE REVIEW

File: src/two_captcha_client.js [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+  }
+
+  /**
+   * @deprecated /load.php route is returning error 500
+   * Get current load from 2Captcha service
+   *
+   * @return {Promise<string>} Promise for an XML containing current load from
+   * 2Captcha service
+   */
+  async load() {
```
Comment (cezar-lima) [inline src/two_captcha_client.js]: Why adding if it is deprecated?

File: src/two_captcha_client.js [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    let res = await this._request('load', 'get');
```
Comment (cezar-lima) [inline src/two_captcha_client.js]:
```suggestion
    return await this._request('load', 'get');
```

File: src/two_captcha_client.js [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+   * @param  {string} captchaId The id of the incorrectly solved captcha
+   * @return {Promise<Boolean>} Promise for a boolean informing if the report
+   * was received
+   */
+  async report(captchaId) {
+    let res = await this._request('res', 'get', {
+      action: 'reportbad',
+      id: captchaId
+    });
+    return res == 'OK_REPORT_RECORDED';
```
Comment (cezar-lima) [inline src/two_captcha_client.js]: Prefer `===`

File: docs/Captcha.html [hunk 4]
```diff
@@ -0,0 +1,580 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (adrianodennanni) [reply docs/Captcha.html]: Auto-generated docs from JSDoc

File: src/constants.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+module.exports = Object.freeze({
+  userAgent: `NodeTwoCaptcha - JavaScript/Node.js package v${require('../package.json').version}`,
```
[the remark this answers is not in the document]
Comment (adrianodennanni) [reply src/constants.js]: Just keeping the same pattern from the Ruby TwoCaptcha gem.

File: src/http_request.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
+'use strict';
+let axios = require('axios');
+const constants = require('./constants');
+const querystring = require('querystring')
+
+const requestUrl = 'http://2captcha.com/in.php?';
+const responseUrl = 'http://2captcha.com/res.php?';
+
+// Uncomment this to log HTTP requests
```
[the remark this answers is not in the document]
Comment (adrianodennanni) [reply src/http_request.js]: Forgot to remove.

File: src/two_captcha_client.js [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+   * Constructor for the 2Captcha client object
+   *
+   * @param  {string}  key                Your 2Captcha API key
+   * @param  {Object}  [params]           Params for the client
+   * @param  {number}  [params.timeout]     milliseconds before giving up on an captcha
+   * @param  {number}  [params.polling]     milliseconds between polling for answer
+   * @param  {Boolean} [params.throwErrors] Whether the client should throw errors or just log the errors
+   * @return {TwoCaptchaClient}           The client object
+   */
+  constructor(key, {
```
[the remark this answers is not in the document]
Comment (adrianodennanni) [reply src/two_captcha_client.js]: Defaults to `{}`

File: src/two_captcha_client.js (hunk 1 above)
Comment (adrianodennanni) [reply src/two_captcha_client.js]: Just keeping the same pattern from the Ruby TwoCaptcha gem. Also, I don't know if this route will work again some day.

**Pull request merged**